### PR TITLE
Move ResponseCacheFeature to new Abstractions package

### PR DESCRIPTION
**Pull request by JunTaoLuo**
cc @Tratcher

### CODE REVIEW

File: src/Microsoft.AspNetCore.ResponseCaching.Abstractions/ResponseCacheHttpContextExtensions.cs [hunk 1]
```diff
@@ -7,9 +7,9 @@ namespace Microsoft.AspNetCore.ResponseCaching
 {
     public static class ResponseCacheHttpContextExtensions
     {
-        public static ResponseCacheFeature GetResponseCacheFeature(this HttpContext httpContext)
+        public static IResponseCacheFeature GetResponseCacheFeature(this HttpContext httpContext)
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.ResponseCaching.Abstractions/ResponseCacheHttpContextExtensions.cs]: Lets remove this extension.

File: src/Microsoft.AspNetCore.ResponseCaching.Abstractions/project.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+﻿{
+  "version": "0.1.0-*",
+  "buildOptions": {
+    "warningsAsErrors": true,
+    "keyFile": "../../tools/Key.snk",
+    "nowarn": [
+      "CS1591"
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.ResponseCaching.Abstractions/project.json]: missing doc comments? Fix it

File: src/Microsoft.AspNetCore.ResponseCaching.Abstractions/project.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      "url": "git://github.com/aspnet/ResponseCaching"
+    },
+    "tags": [
+      "aspnetcore",
+      "cache",
+      "caching"
+    ]
+  },
+  "dependencies": {
+    "Microsoft.AspNetCore.Http.Abstractions": "1.1.0-*"
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.ResponseCaching.Abstractions/project.json]: Your dependencies will be reduced to Extensions.Primities

**Comment by JunTaoLuo** [comment]
:up::date:

**Comment by Tratcher** [comment]
:shipit:

File: src/Microsoft.AspNetCore.ResponseCaching/project.json [hunk 4]
```diff
@@ -24,6 +24,7 @@
   "dependencies": {
     "Microsoft.AspNetCore.Http": "1.1.0-*",
     "Microsoft.AspNetCore.Http.Extensions": "1.1.0-*",
+    "Microsoft.AspNetCore.ResponseCaching.Abstractions": "0.1.0-*",
```
Comment (jkotalik) [inline src/Microsoft.AspNetCore.ResponseCaching/project.json]: Curious why this is 0.1.0?

**Comment by Tratcher** [comment]
Was the version number change approved by Murat/Eilon/Damian? It's hard to go backwards later.

**Comment by JunTaoLuo** [comment]
I spoke to @muratg who approved it. We will definitely release it as 1.0.0. The only point of contention is whether it will be LTS or FTS.

**Pull request merged**